sdk/messaging/azeventhubs/internal/utils/retrier.go

```diff
@@ -34,7 +34,7 @@ func (rf *RetryFnArgs) ResetAttempts() {
 
 // Retry runs a standard retry loop. It executes your passed in fn as the body of the loop.
 // It returns if it exceeds the number of configured retry options or if 'isFatal' returns true.
-func Retry(ctx context.Context, eventName log.Event, operation string, o exported.RetryOptions, fn func(ctx context.Context, callbackArgs *RetryFnArgs) error, isFatalFn func(err error) bool) error {
+func Retry(ctx context.Context, eventName log.Event, prefix func() string, o exported.RetryOptions, fn func(ctx context.Context, callbackArgs *RetryFnArgs) error, isFatalFn func(err error) bool) error {
```

Comment:
Why does this need to be a func now?

Reply:
So it can be dynamic - for instance, if we rotate out the link and connection ID in between retries it's good to update the logging prefix as well.

Reply:
Example here: https://github.com/Azure/azure-sdk-for-go/pull/19683/files/bd9cc2a0e0f4aeb3dccb3f0a754681431bab0079#diff-5f99975d3c585da6e7dfb4726eca7c769eff91f91d10eaa57a0e6c69d12e30f0R170